bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -458,26 +502,14 @@ void bsg_serialize_breadcrumbs(const bugsnag_event *event, JSON_Array *crumbs) {
       JSON_Value *crumb_val = json_value_init_object();
       JSON_Object *crumb = json_value_get_object(crumb_val);
       json_array_append_value(crumbs, crumb_val);
+
       bugsnag_breadcrumb breadcrumb = event->breadcrumbs[current_index];
       json_object_set_string(crumb, "name", breadcrumb.name);
       json_object_set_string(crumb, "timestamp", breadcrumb.timestamp);
       json_object_set_string(crumb, "type",
                              bsg_crumb_type_string(breadcrumb.type));
-
-      JSON_Value *meta_val = json_value_init_object();
-      JSON_Object *meta = json_value_get_object(meta_val);
-      json_object_set_value(crumb, "metaData", meta_val);
-      int metadata_index = 0;
-      while (strlen(breadcrumb.metadata[metadata_index].key) > 0) {
-        json_object_set_string(meta, breadcrumb.metadata[metadata_index].key,
-                               breadcrumb.metadata[metadata_index].value);
-        metadata_index++;
-      }
-
+      bsg_serialize_breadcrumb_metadata(breadcrumb.metadata, crumb);
       current_index++;
-      if (current_index == BUGSNAG_CRUMBS_MAX) {
-        current_index = 0;
-      }
```

Comment:
I would have thought this is still required in order to good back to the start of an array (as it's a circular structure)?

Reply:
This is required - added back in.